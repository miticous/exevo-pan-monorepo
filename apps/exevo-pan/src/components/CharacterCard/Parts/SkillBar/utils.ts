export const getDecimalPart = (float: number): number =>
  Math.round((float % 1) * 100)
