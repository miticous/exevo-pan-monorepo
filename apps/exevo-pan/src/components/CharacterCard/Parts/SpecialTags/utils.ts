import { OutfitCheck } from './types'

const CHARM_CHECK = 7
const QUEST_CHECK = 26
const MOUNT_CHECK = 35
const STORE_CHECK = 4
const rareMounts = new Set<string>([
  'Rift Runner',
  'Phantasmal Jade',
  'Singeing Steed',
  'Neon Sparkid',
  'Vortexion',
  'Phant',
  'Antelope',
])

const outfitChecks: Record<string, OutfitCheck> = {
  Mage: {
    test: ({ type }, characterObject) => !characterObject.sex && type >= 2,
    tag: "Ferumbras' Hat",
  },
  Summoner: {
    test: ({ type }, characterObject) => characterObject.sex && type >= 2,
    tag: "Ferumbras' Hat",
  },
  Elementalist: {
    tag: 'Elementalist addons',
    test: ({ type }) => type > 0,
  },
  'Golden Outfit': {
    tag: 'Golden Outfit',
    test: () => true,
  },
  Revenant: {
    tag: 'Revenant addons',
    test: ({ type }) => type > 0,
  },
}

export const getCharacterTags = (character: CharacterObject): string[] => {
  const { charms, quests, mounts, outfits, storeMounts, storeOutfits } =
    character

  const tags: string[] = []

  if (charms.length >= CHARM_CHECK) tags.push('manyCharms')
  if (quests.length >= QUEST_CHECK) tags.push('manyQuests')
  if (mounts.length >= MOUNT_CHECK) tags.push('manyMounts')
  if (storeOutfits.length + storeMounts.length >= STORE_CHECK) {
    tags.push('manyStoryCosmetics')
  }

  mounts.forEach((mount) => {
    if (rareMounts.has(mount)) tags.push(mount)
  })

  outfits.forEach((outfit) => {
    const check = outfitChecks[outfit.name]
    if (check?.test(outfit, character)) tags.push(check.tag)
  })

  return tags
}
