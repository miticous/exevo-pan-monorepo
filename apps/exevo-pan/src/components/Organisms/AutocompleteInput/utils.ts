export const filterByTerm = (term: string, array: Option[]): Option[] => {
  const regex = new RegExp(term, 'i')
  return array.filter((item) => regex.test(item.name))
}

export function circularArrayIndex<T>(index: number, array: Array<T>): number {
  const maxIndex = array.length - 1
  if (index < 0) {
    return maxIndex
  }
  if (index > maxIndex) {
    return 0
  }

  return index
}
