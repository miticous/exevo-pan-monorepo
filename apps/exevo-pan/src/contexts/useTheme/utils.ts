import Themes from 'styles/themes'
import { localStorageKeys } from 'Constants'

export const getInitialTheme = (): string => {
  if (typeof window !== 'undefined') {
    const persistedColorPreference = window.localStorage.getItem(
      localStorageKeys.THEME_DATA,
    )
    if (persistedColorPreference && Themes[persistedColorPreference]?.title)
      return persistedColorPreference

    const browserDefault = window.matchMedia('(prefers-color-scheme: dark)')
    if (browserDefault?.matches) return 'dark-theme'
  }

  return Themes.default.title
}

export const injectCssVariables = (themeName: keyof typeof Themes): void => {
  const { colors } = Themes[themeName]
  const root = document.documentElement

  Object.keys(colors).forEach((key) => {
    root.style.setProperty(`--${key}`, colors[key as keyof typeof colors])
  })

  document
    .querySelector('meta[name=theme-color]')
    ?.setAttribute('content', colors.primary)

  document
    .querySelector('meta[name=msapplication-navbutton-color]')
    ?.setAttribute('content', colors.primary)
}
