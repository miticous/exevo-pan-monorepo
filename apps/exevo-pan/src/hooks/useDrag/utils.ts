import { MouseTouchEvent, positionObject } from './types'

export const stopBubbling = (event: MouseTouchEvent): void => {
  event.stopPropagation()
}

const getNativeEvent = (event: MouseTouchEvent) =>
  event.nativeEvent instanceof MouseEvent
    ? event.nativeEvent
    : event.nativeEvent.changedTouches[0]

export const getPosition = (event: MouseTouchEvent): positionObject => {
  const nativeEvent = getNativeEvent(event)

  const target = nativeEvent.target as HTMLDivElement
  const parentNode = target.parentNode as HTMLDivElement

  const { left, top } = parentNode.getBoundingClientRect()
  const { clientX, clientY } = nativeEvent
  return {
    x: clientX - left,
    y: clientY - top,
  }
}
