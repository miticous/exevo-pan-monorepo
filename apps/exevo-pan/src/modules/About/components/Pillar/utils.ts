export const generateNavId = (id?: string): string => `${id}-nav`
