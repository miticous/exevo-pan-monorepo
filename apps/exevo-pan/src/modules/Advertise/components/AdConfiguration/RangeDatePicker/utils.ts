const formatDate = (timestamp: number): string => {
  const date = new Date(timestamp)
  const day = date.getDate().toString()
  const month = (date.getMonth() + 1).toString()
  const year = date.getFullYear().toString()

  return `${month}/${day}/${year}`
}

export const getDaysUntilAuctionEnd = (auctionEnd?: number): string[] => {
  if (!auctionEnd) return []

  const formatedEndDate = formatDate(auctionEnd * 1000)
  const formatedCurrentDate = formatDate(+new Date())

  const endDate = new Date(formatedEndDate)
  const currentDate = new Date(formatedCurrentDate)
  const daysInRange = [formatedCurrentDate]
  while (currentDate < endDate) {
    currentDate.setDate(currentDate.getDate() + 1)
    daysInRange.push(formatDate(+currentDate))
  }

  return daysInRange
}

export const partitionByMonths = (dateArray: string[]): string[][] => {
  const monthsPartition = new Set<string>([])

  dateArray.forEach((dateString) => {
    const [month] = dateString.split('/')
    monthsPartition.add(month)
  })

  return [...monthsPartition].map((month) =>
    dateArray.filter((dateString) => {
      const [dateStringMonth] = dateString.split('/')

      return month === dateStringMonth
    }),
  )
}

export const fillWithDays = (
  initialDateString: string,
  amount: number,
  step: number,
): string[] => {
  const currentDate = new Date(initialDateString)

  return Array.from(
    { length: amount },
    () => +currentDate.setDate(currentDate.getDate() + step),
  )
    .sort((a, b) => a - b)
    .map(formatDate)
}

export const getWeekdayNumber = (date: string): number =>
  new Date(date).getDay()

export const getDay = (date: string): number => new Date(date).getDate()

export const getMonth = (date: string): number => new Date(date).getMonth()
