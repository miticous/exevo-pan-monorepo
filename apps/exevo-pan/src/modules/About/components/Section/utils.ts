import { copyToClipboard } from 'utils'

export const CopyToClipboard = (id: string): void =>
  copyToClipboard(
    `https://${window.location.hostname}${window.location.pathname}#${id}`,
  )
