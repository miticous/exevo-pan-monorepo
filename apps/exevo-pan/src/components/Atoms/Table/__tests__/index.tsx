import { screen } from '@testing-library/react'
import { renderWithProviders } from 'utils/test'
import Table from '..'

describe('<Table />', () => {
  test('should render all elements correctly', () => {
    renderWithProviders(
      <Table>
        <div role="none" />
        <Table.Element>
          <Table.Head>
            <Table.Row>
              <Table.HeadColumn>#</Table.HeadColumn>
              <Table.HeadColumn>Nickname</Table.HeadColumn>
            </Table.Row>
          </Table.Head>

          <Table.Body>
            <Table.Row>
              <Table.Column>1</Table.Column>
              <Table.Column>Ksu</Table.Column>
            </Table.Row>
            <Table.Row>
              <Table.Column>2</Table.Column>
              <Table.Column>Nana</Table.Column>
            </Table.Row>
          </Table.Body>
        </Table.Element>
      </Table>,
    )

    expect(screen.getByRole('table')).toBeInTheDocument()
    const [firstHeadColumn, secondHeadColumn] =
      screen.getAllByRole('columnheader')
    expect(firstHeadColumn).toHaveTextContent('#')
    expect(secondHeadColumn).toHaveTextContent('Nickname')

    expect(screen.getAllByRole('rowgroup')).toHaveLength(2)

    const rows = screen.getAllByRole('row')
    expect(rows).toHaveLength(3)

    const [, firstTableRow, secondTableRow] = rows
    expect(firstTableRow).toHaveTextContent('1')
    expect(firstTableRow).toHaveTextContent('Ksu')
    expect(secondTableRow).toHaveTextContent('2')
    expect(secondTableRow).toHaveTextContent('Nana')

    expect(screen.getByRole('none')).toBeInTheDocument()
  })

  test('should display card head correctly with/without subtitles', () => {
    const { rerender } = renderWithProviders(
      <Table title="Main title">content</Table>,
    )

    const titleElement = screen.getByRole('heading')
    expect(titleElement).toHaveTextContent('Main title')

    rerender(
      <Table title="Main title" subtitle="subtitle content">
        content
      </Table>,
    )

    expect(titleElement).toHaveTextContent('Main title')
    expect(screen.getByText('subtitle content')).toBeInTheDocument()
  })

  test('caption element should NOT be visible', () => {
    renderWithProviders(
      <Table title="Main title">
        <div>visible content</div>
        <caption>hidden content</caption>
      </Table>,
    )

    expect(screen.getByText('visible content')).toBeInTheDocument()
    expect(screen.queryByRole('caption')).not.toBeInTheDocument()
  })

  test('head column highlight prop should display asc/desc symbol', () => {
    const { container, rerender } = renderWithProviders(
      <Table>
        <Table.Element>
          <Table.Head>
            <Table.Row>
              <Table.HeadColumn highlighted>#</Table.HeadColumn>
              <Table.HeadColumn>Nickname</Table.HeadColumn>
            </Table.Row>
          </Table.Head>
        </Table.Element>
      </Table>,
    )

    expect(container).toMatchSnapshot()

    rerender(
      <Table>
        <Table.Element>
          <Table.Head>
            <Table.Row>
              <Table.HeadColumn highlighted desc>
                #
              </Table.HeadColumn>
              <Table.HeadColumn>Nickname</Table.HeadColumn>
            </Table.Row>
          </Table.Head>
        </Table.Element>
      </Table>,
    )

    expect(container).toMatchSnapshot()
  })
})
