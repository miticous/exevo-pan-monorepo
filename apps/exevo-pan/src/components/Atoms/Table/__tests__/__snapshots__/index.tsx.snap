// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> head column highlight prop should display asc/desc symbol 1`] = `
<div>
  <section
    class="styles__Wrapper-sc-127r6c8-0 dDEOTP"
  >
    <table
      class="styles__Table-sc-127r6c8-3 eciGZz"
    >
      <thead
        class="styles__Head-sc-127r6c8-5 llrXvQ"
      >
        <tr
          class="styles__Row-sc-127r6c8-4 bQMJNc"
        >
          <th
            class="styles__HeadColumn-sc-127r6c8-6 iVOZZv"
          >
            #
          </th>
          <th
            class="styles__HeadColumn-sc-127r6c8-6 iPuRde"
          >
            Nickname
          </th>
        </tr>
      </thead>
    </table>
  </section>
</div>
`;

exports[`<Table /> head column highlight prop should display asc/desc symbol 2`] = `
<div>
  <section
    class="styles__Wrapper-sc-127r6c8-0 dDEOTP"
  >
    <table
      class="styles__Table-sc-127r6c8-3 eciGZz"
    >
      <thead
        class="styles__Head-sc-127r6c8-5 llrXvQ"
      >
        <tr
          class="styles__Row-sc-127r6c8-4 bQMJNc"
        >
          <th
            class="styles__HeadColumn-sc-127r6c8-6 jYlGil"
          >
            #
          </th>
          <th
            class="styles__HeadColumn-sc-127r6c8-6 iPuRde"
          >
            Nickname
          </th>
        </tr>
      </thead>
    </table>
  </section>
</div>
`;
